src/testers.jl

```diff
@@ -100,17 +100,20 @@ function test_frule(
     isapprox_kwargs = (; rtol=rtol, atol=atol, kwargs...)
 
     @testset "test_frule: $f on $(_string_typeof(args))" begin
-        xẋs = auto_primal_and_tangent.(args)
-        xs = primal.(xẋs)
-        tangents = (rand_tangent(f), tangent.(xẋs)...)
-        if check_inferred && _is_inferrable(f, deepcopy(xs)...; deepcopy(fkwargs)...)
-            _test_inferred(frule, (deepcopy(tangents)...,), f, deepcopy(xs)...; deepcopy(fkwargs)...)
+
+        primals_and_tangents = auto_primal_and_tangent.((f, args...))
+        func, xs = Iterators.peel(primal.(primals_and_tangents))
+        tangents = tangent.(primals_and_tangents)
+
+        if check_inferred && _is_inferrable(func, deepcopy(xs)...; deepcopy(fkwargs)...)
```

Comment:
note that we need to deep copy `func` here also.
 as well as on line 109, 112, and 116
 
 Possibly this also could use `_is_inferable(call_on_copy(primals...))` ?

Reply:
Not sure how this would work? Otherwise it's ready for re-review

Reply:
Ah year it wouldn't quite work